python/graphstorm/gconstruct/utils.py

```diff
@@ -38,11 +38,19 @@
 PICKLE_CROSS_PROCESS_STORAGE = "pickle"
 EXT_MEMORY_STORAGE = "ext_memory"
 
+def _is_numeric(arr):
+    """ Check if the input array has the numeric data type.
+    """
+    return np.issubdtype(arr.dtype, np.number) or arr.dtype == bool
+
 def _to_ext_memory(name, data, path):
     if isinstance(data, np.ndarray):
         assert name is not None
         path = os.path.join(path, f"{name}.npy")
-        if len(data) > 0:
+        # We only save a data array with numeric or boolean values to the disk.
+        # This avoids the problem of saving an array of objects to disks.
+        # Note: There is a bug in Numpy when saving an array of objects to disks.
+        if len(data) > 0 and _is_numeric(data):
```

Comment:
Why not check whether data has dtype numpy.object?

Reply:
there are many other numpy dtypes, such as str, that may not work well with external memory.